### PR TITLE
[Merged by Bors] - chore: golf proofs about `IsBoundedBilinearMap`

### PR DESCRIPTION
**Pull request by urkud**
Add `IsBoundedBilinearMap.toContinuousLinearMap` and use it to golf
proofs by reusing facts about bundled bilinear maps
`E →L[𝕜] F →L[𝕜] G`.

### All changes

- Add `IsBoundedBilinearMap.toContinuousLinearMap`.
- Rename `IsBoundedBilinearMap.is_O'` to
  `IsBoundedBilinearMap.isBigO'`.
- Rename `isBoundedBilinearMap_deriv_coe` to
  `IsBoundedBilinearMap.deriv_apply`.
- Add `LinearMap.mkContinuousOfExistsBound₂`, a bilinear map version
  of `LinearMap.mkContinuousOfExistsBound` and use it to redefine
  `LinearMap.mkContinuous₂`. The new definition is definitionally
  equal to the old one.
- Import `Mathlib.Analysis.NormedSpace.OperatorNorm` instead of
  `Mathlib.Analysis.NormedSpace.BoundedLinearMaps` in
  `Mathlib.Analysis.Calculus.FDeriv.Basic`.
- Golf many proofs

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by mcdoll — APPROVED** [review]
Looks good to me

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 1]
```diff
@@ -717,25 +717,35 @@ theorem mkContinuous_norm_le' (f : E →ₛₗ[σ₁₂] F) {C : ℝ} (h : ∀ x
...
+        rw [ContinuousLinearMap.smul_apply, mkContinuousOfExistsBound_apply,
+          mkContinuousOfExistsBound_apply, map_smulₛₗ, smul_apply] } <|
+    let ⟨C, hC⟩ := h; ⟨max C 0, norm_mkContinuous₂_aux f C hC⟩
+
+/-- Create a bilinear map (represented as a map `E →L[𝕜] F →L[𝕜] G`) from the corresponding linear
+map and a bound on the norm of the image. The linear map can be constructed using
+`LinearMap.mk₂`. -/
+def mkContinuous₂ (f : E →ₛₗ[σ₁₃] F →ₛₗ[σ₂₃] G) (C : ℝ) (hC : ∀ x y, ‖f x y‖ ≤ C * ‖x‖ * ‖y‖) :
+    E →SL[σ₁₃] F →SL[σ₂₃] G :=
+  mkContinuousOfExistsBound₂ f ⟨C, hC⟩
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:748]: Why do we want both versions of this definition? Is the any advantage to bundling the existential?

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:748]: Yes, this way it's clear that data does not depend on C. See the file about bounded (bi)linear map predicate for an example.

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:748]: @eric-wieser Do you have any more objections? If not, can we merge it?

Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:748]: Can you add a comment explaining what you said above? I assume `TODO: remove after the port is done` would be accurate?

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:748]: What do you want to remove after the port is done?

**Comment by fpvandoorn** [comment]
bors merge

**Pull request closed without merging**